remote/analyze_handler.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python2
+
+import os
+import logging
+import subprocess
+import zipfile
+import redis
+import sched
+import time
+
+from enum import Enum
+
+LOGGER = logging.getLogger('ANALYZE HANDLER')
+LOGGER.setLevel(logging.INFO)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter(
+    '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+LOGGER.addHandler(ch)
+
+
+class AnalyzeStatus(Enum):
+    ID_PROVIDED = 'ID_PROVIDED'
+    QUEUED = 'QUEUED'
+    ANALYZE_IN_PROGRESS = 'ANALYZE_IN_PROGRESS'
+    ANALYZE_COMPLETED = 'ANALYZE_COMPLETED'
+
+
+def preAnalyze(analyze_id, part_number):
+    LOGGER.info('Pre Analyze Step')
+
+    file_name = 'source'
+    file_extension = '.zip'
+
+    file_path = os.path.join('/workspace', analyze_id)
+
+    source_file_without_extension = file_name + '_' + str(part_number)
+    source_file_with_extension = source_file_without_extension + file_extension
+
+    zip_file_path = os.path.join(file_path, source_file_with_extension)
+
+    analyze_dir_path = os.path.join(file_path, source_file_without_extension)
+
+    with zipfile.ZipFile(zip_file_path) as zf:
```

Comment:
Shouldn't this function return in case of some error. If extracting the files or getting the build command fails what can be done?
Maybe some extra analyze status values would be nice for these cases:
 - `preanalyze_failed`
- `analyze_failed` 
- `postanalyze_failed`

Reply:
Good idea, will look into that, thanks!